src/charm.py

```diff
@@ -371,15 +364,20 @@ def _on_update_status(self, event):
         # being able to start the process at all. If this is the case we need to restart the
         # process ourselves. We defer to give it time to get started before reporting its
         # status.
-        with MongoDBConnection(self.mongodb_config, "localhost", direct=True) as direct_mongo:
-            if not direct_mongo.is_ready:
-                logger.debug("mongodb service is not ready yet, restarting.")
-                self.restart_mongod_service()
-                # ensure that the correct port is open for the service
-                self._open_port_tcp(self._port)
-                self.unit.status = WaitingStatus("Waiting for MongoDB to start")
-                event.defer()
-                return
+        try:
+            with MongoDBConnection(self.mongodb_config, "localhost", direct=True) as direct_mongo:
+                if not direct_mongo.is_ready:
+                    logger.debug("mongodb service is not ready yet, restarting.")
+                    self.restart_mongod_service()
```

Comment:
I don't think it is needed anymore if the snap unit has `restart-condition=always` and `restart-delay=1m`